### PR TITLE
Make numThreads unsigned

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: include/swift/AST/SILOptions.h [hunk 1]
```diff
@@ -41,7 +41,7 @@ class SILOptions {
   int UnrollThreshold = 250;
 
   /// The number of threads for multi-threaded code generation.
-  int NumThreads = 0;
+  unsigned int NumThreads = 0;
```
Comment (CodaFi) [inline include/swift/AST/SILOptions.h:44]: Just `unsigned`.

**Comment by CodaFi** [comment]
In general, I'm not a fan of these kinds of arguments for introducing unsignedness. If there's an invalid state we need to make unrepresentable, it should be handled at the level of the driver warning the user that e.g. they passed the wrong number of threads.

**Comment by pi1024e** [comment]
Except every function the variable itself is used in has an unsigned parameter called numThreads.

**Comment by pi1024e** [comment]
Also if an invalid number is entered, we have assert statements, catch-throws, etc.

**Comment by CodaFi** [comment]
Right, and those asserts will be compiled out in release builds, so the only thing we've really shored up here is the implicit conversion from signed to unsigned. The usual exceptions for a signed-to-unsigned change involve wrapping arithmetic, bitlayout concerns, or anxiety about representable value range.

**Comment by CodaFi** [comment]
@swift-ci smoke test

**Comment by pi1024e** [comment]
Wait, curious, are these smoke tests being run on a dev build of Big Sur? or the current stable macOS

**Comment by CodaFi** [comment]
The exact OS install on the bots doesn't really matter. They are all building with the Xcode 12 beta and SDK. I am still able to build locally with macOS 10.15.

**Comment by pi1024e** [comment]
Thank you

File: lib/IRGen/IRGen.cpp [hunk 2]
```diff
@@ -1111,7 +1111,7 @@ struct LLVMCodeGenThreads {
 /// All this is done in multiple threads.
 static void performParallelIRGeneration(
     const IRGenOptions &Opts, swift::ModuleDecl *M, std::unique_ptr<SILModule> SILMod,
-    StringRef ModuleName, int numThreads,
+    StringRef ModuleName, unsigned int numThreads,
```
Comment (CodaFi) [inline lib/IRGen/IRGen.cpp:1114]: Come to think of it, let's drop this parameter. You can get it from the options in the SIL module.

**Comment by pi1024e** [comment]
@CodaFi done

**Comment by CodaFi** [comment]
@swift-ci smoke tes

**Comment by CodaFi** [comment]
@swift-ci smoke test

**Comment by pi1024e** [comment]
@CodaFi the original test failed because of some build issues unrelated to the code itself. can you do one more smoke test? I rebased just in case too

**Comment by CodaFi** [comment]
You don't need to rebase for unrelated issues. CI will automatically fast-forward before applying your changes.

**Comment by CodaFi** [comment]
@swift-ci please smoke tes

**Comment by CodaFi** [comment]
@swift-ci please smoke test

**Comment by CodaFi** [comment]
@swift-ci please smoke test

**Comment by CodaFi** [comment]
@swift-ci please smoke test

**Comment by pi1024e** [comment]
@DougGregor what do you think?

**Comment by CodaFi** [comment]
@swift-ci please smoke test

**Comment by davidungar** [comment]
Yes, but what if we get something that can produce source code from an intermediate representation? Wouldn't that constitute a thread with number -1? ;)

**Pull request closed without merging**